packages/fxa-settings/src/components/LinkExpiredResetPassword/index.tsx

```diff
@@ -12,27 +12,28 @@ import { LinkExpired } from '../LinkExpired';
 type LinkExpiredResetPasswordProps = {
   email: string;
   viewName: string;
+  integration: Integration;
 };
 
 export const LinkExpiredResetPassword = ({
   email,
   viewName,
+  integration,
 }: LinkExpiredResetPasswordProps) => {
   // TODO in FXA-7630 add metrics event and associated tests for users hitting the LinkExpired page
   const account = useAccount();
-  const relier = useRelier();
 
   const [resendStatus, setResendStatus] = useState<ResendStatus>(
     ResendStatus['not sent']
   );
 
   const resendResetPasswordLink = async () => {
```

Comment:
Perhaps this comment is overthought, but if this function was passed into LinkExpiredResetPassword, like it is for the LinkExpired component, we'd remove the dependency on intergration and account classes in this component. IMO, this would be a better abstraction and make this LinkValidator cleaner. e.g. You could hoist the logic for resending the reset password link up a few levels, and decrease the amount of context and stuff you have to pass around. It also seems like the logic for resending the link could live here, but it could also just as easily be defined elsewhere and passed in. I think doing so would make the link validator cleaner. I have left a related comment in the LinkValidator component.